cstar_ocean/input_dataset.py

```diff
@@ -48,17 +49,23 @@ def __init__(
         base_model: BaseModel
             The base model with which this input dataset is associated
         source: str
-            URL pointing to the netCDF file containing this input dataset
+            URL or path pointing to the netCDF file containing this input dataset
         file_hash: str
-            The 256 bit SHA sum associated with the file for verifying downloads
+            The 256 bit SHA sum associated with the file for verification
 
         """
 
         self.base_model: BaseModel = base_model
+
         self.source: str = source
         self.file_hash: str = file_hash
+
         self.exists_locally: Optional[bool] = None
         self.local_path: Optional[str] = None
+        if _get_source_type(source) == "path":
+            self.exists_locally = True
+            self.local_path = source
```

Comment:
Again I wonder if this could all be abstracted behind a class. e.g.
```python
self.source = DataSource(source)
```
then later use `self.source.exists_locally` and `self.source.path`.

Reply:
Yes, I began implementing this for this PR but it got hairier than expected. I think it requires a lot of design considerations and should be returned to once the core feature set is complete. `AdditionalCode` and `BaseModel` also have a `source_repo` attribute that could be replaced with an instance of whatever this class will be. I think this could be part of the tidy-up that brings in `pathlib`. I'll raise an issue for now.